js/src/carousel.js

```diff
@@ -390,10 +390,12 @@ const Carousel = (($) => {
 
         if (typeof config === 'number') {
           data.to(config)
-
-        } else if (action) {
-          data[action]()
-
+        } else if (typeof action === 'string') {
+          try {
```

Comment:
It's possible for the method itself to throw an exception, which shouldn't be caught and misidentified as a nonexistent method.

So this should instead be something like:
```js
let method = data[action]
if (method === undefined) {
  // throw an error
}